### ISSUE TITLE
ParamSpec type check doesn't seem to work

### ISSUE THREAD

**Issue by ksato9700** [microsoft/pyright#1410]
**Describe the bug**
I use ParamSpec, which will be introduced in 3.10, to check the parameter types of function calls.  See the sample code below to see what I tried.

I believe the call of `call_callback` in the last line of the code should have been a type error by pyright because the type of parameter of the given function (`[str]`) is not equal to the type of given parameters to the `call_callback` (`[str, int]`). However, I only get a warning from pyright
```
  5:31 - warning: TypeVar "P" appears only once in generic function signature (reportInvalidTypeVarUse)
```

I'm not sure if Is is a wrong way to use ParamSpec, or a bug of pyright, but let me report it to verify that.

**To Reproduce**
```
$ pyright cb.py
```

**Expected behavior**
I'd like pyright to emit an error of the wrong parameter use.

**Sample code**

```python cb.py
from typing import Callable, ParamSpec

P = ParamSpec("P")

def call_callback(f: Callable[P, str], *args: P.args, **kwargs: P.kwargs) -> str:
  return f(*args, **kwargs)

def reversed_str(s: str) -> str:
  return "".join(reversed(s))

def multi_str(s: str, i: int) -> str:
  return s * i

print(call_callback(reversed_str, "abc")) # "cba" is printed
print(call_callback(multi_str, "abc", 2)) # "abcabc"is printed
print(call_callback(reversed_str, "abc", 3)) # run-time error
```

**VS Code extension or command-line**
I used command line to test, with the following configuration.

```
{
	"pythonversion": "3.10"
}
```

**Additional context**
pyright version: 1.1.104
python version: 3.10.0a4+ (pyenv "3.10-dev" as of 2021.01.24)

**Comment by YodaEmbedding** [comment]
```python
def call_callback(f: Callable[P, str], *args: P.args, **kwargs: P.kwargs) -> str:
```

PEP-612 seems to intend that `P` is referenced by some enclosing decorator function before `P.args` is referenced in its inner function. Example:

```python
def decorator(f: Callable[P, str]) -> Callable[P, str]:
    def inner(*args: P.args, **kwargs: P.kwargs) -> str:
        return ""
    return inner
```

**Comment by erictraut** [comment]
You are correct. When I read PEP 612, I missed the fact that `P.args` and `P.kwargs` can be used within the signature of the function that uses `Callable[P, X]`. Upon closer reading, I see that the PEP includes an example of this:

```python
def twice(f: Callable[P, int], *args: P.args, **kwargs: P.kwargs) -> int:
  return f(*args, **kwargs) + f(*args, **kwargs)
```

I would normally use mypy to confirm my assumptions, but mypy (and to my knowledge, no other Python type checkers other than pyright) have implemented PEP 612 yet.

This will require a bit of work to address.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by ksato9700** [comment]
@YodaEmbedding @erictraut 
Thank you for your messages. I understand PEP-612 use the decorator use case to explain the feature, but I wanted to use a simpler example for my blog post. The sample code is a little bit artificial, but I'm happy to hear that it is a right use of the ParamSpec. Thank you very much for your support!!

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.105, which I just published. It will also be included in the next release of pylance.
